nodes/node_duplicator.lua

```diff
@@ -60,90 +60,103 @@ minetest.register_node("digtron:duplicator", {
 	selection_box = {
 	    type = "regular"
 	},
-	
+
 	on_construct = function(pos)
 		local meta = minetest.get_meta(pos)
 		meta:set_string("formspec", inventory_formspec_string)
 		local inv = meta:get_inventory()
 		inv:set_size("main", 8*4)
 	end,
-	
+
 	can_dig = function(pos,player)
 		local meta = minetest.get_meta(pos)
 		local inv = meta:get_inventory()
 		return inv:is_empty("main")
 	end,
-	
+
 	allow_metadata_inventory_put = function(pos, listname, index, stack, player)
 		if minetest.get_item_group(stack:get_name(), "digtron") > 0 then
 			return stack:get_count()
 		else
 			return 0
 		end
 	end,
-	
+
 	on_receive_fields = function(pos, formname, fields, sender)
+		local player_name = sender:get_player_name()
 		if fields.help then
-			minetest.after(0.5, doc.show_entry, sender:get_player_name(), "nodes", "digtron:duplicator", true)
+			minetest.after(0.5, doc.show_entry, player_name, "nodes", "digtron:duplicator", true)
+			return
 		end
-	
+
 		if fields.duplicate then
 			local node = minetest.get_node(pos)
 			local meta = minetest.get_meta(pos)
 			local inv = meta:get_inventory()
 			local target_pos = vector.add(pos, minetest.facedir_to_dir(node.param2))
 			local target_node = minetest.get_node(target_pos)
+			local target_name
 
-			if target_node.name ~= "digtron:empty_crate" then
+			if "digtron:empty_crate" == target_node.name then
```

Comment:
Minor: Inconsistent style (compared with the rest of this mod). The constant is usually written after the variable.